packages/wrangler/src/whoami.tsx

```diff
@@ -89,3 +128,42 @@ type AccountInfo = { name: string; id: string };
 async function getAccounts(): Promise<AccountInfo[]> {
 	return await fetchListResult<AccountInfo>("/accounts");
 }
+
+async function getTokenPermissions(): Promise<string[] | undefined> {
+	// Tokens can either be API tokens or Oauth tokens.
+	// Here we only extract permissions from OAuth tokens.
+
+	return getScopes() as string[];
+
+	// In future we may be able to get the token permissions on an API token,
+	// but currently we cannot as that permission is not able to be added to
+	// an API token.
```

Comment:
This comment and following code block is really awesome additional context and insight in thinking ahead!